pkgs/applications/networking/remote/x2goserver/default.nix

```diff
@@ -0,0 +1,79 @@
+{ stdenv, lib, fetchurl, makeWrapper, perl, perlPackages, which, nx-libs
+, utillinux, coreutils, glibc, gawk, gnused, gnugrep, findutils, xorg
+, nettools, iproute, bc, procps, psmisc, lsof, pwgen, openssh, sshfs
+}:
+
+let binaryDeps = [
+  perl which nx-libs utillinux coreutils glibc.bin gawk gnused gnugrep
+  findutils nettools iproute bc procps psmisc lsof pwgen openssh sshfs
+  xorg.xauth xorg.xinit xorg.xrandr xorg.xmodmap xorg.xwininfo xorg.fontutil
+  xorg.xkbcomp xorg.setxkbmap
+]; in
+stdenv.mkDerivation rec {
+  pname = "x2goserver";
+  version = "4.1.0.3";
+
+  src = fetchurl {
+    url = "http://code.x2go.org/releases/source/x2goserver/${pname}-${version}.tar.gz";
+    sha256 = "1l6wd708kbipib4ldprfiihqmj4895nifg0bkws4x97majislxk7";
+  };
+
+  buildInputs = binaryDeps;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  patches = [ ./xsession-nixos-support.patch ];
+
+  preConfigure = ''
+    patchShebangs .
+    perl Makefile.PL
+    for i in */Makefile; do
+      substituteInPlace "$i" --replace "-o root -g root " ""
+    done
+    substituteInPlace libx2go-server-db-perl/Makefile --replace "chmod 2755" "chmod 755"
+    for i in X2Go/Config.pm x2goserver/sbin/x2godbadmin x2goserver/bin/x2go*
+    do
+      substituteInPlace $i --replace '/etc/x2go' "/var/lib/x2go/conf"
+    done
+    substituteInPlace X2Go/Server/DB.pm \
+      --replace '$x2go_lib_path/libx2go-server-db-sqlite3-wrapper' \
+                '/run/wrappers/bin/x2gosqliteWrapper'
+    substituteInPlace x2goserver/sbin/x2godbadmin --replace 'user="x2gouser"' 'user="x2go"'
+    substituteInPlace X2Go/Server/DB/SQLite3.pm --replace "user='x2gouser'" "user='x2go'"
+  '';
+
+  makeFlags = [ "PREFIX=/" "NXLIBDIR=${nx-libs}/lib/nx" "PERL=${perl}/bin/perl" "PERL_INSTALLDIRS=vendor" ];
```

Comment:
See comment about about potentially replacing the `PERL` binary here with `perl.withPackages` and then not having to deal with any `-I` or `PERL5LIB` mess.

Reply:
It's done, at least most bundled binaries don't need an extra PERL5LIB. Unfortunately this app also bundles it's own perl module, so I had to extract that to an extra derivation, which I can put in the perl.withPackages to avoid circular dependencies.
But it looks a lot cleaner this way in any case.